Makefile

```diff
@@ -124,6 +124,14 @@ test-unused-modules:
 test-deleted-renamed-referenced-modules:
 	tools/test_deleted_renamed_referenced_modules `git diff --name-only --exit-code --diff-filter=DR $$(git merge-base master HEAD) | grep '^tests/*'`
 
+.PHONY: test-deleted-renamed-testdata
+test-deleted-renamed-testdata:
+	tools/test_deleted_renamed_testdata `git diff --name-only --exit-code --diff-filter=DR $$(git merge-base master HEAD) | grep '^test_data/*'`
+
+.PHONY: detect-nonexistent-testdata
+detect-nonexistent-testdata:
+	tools/detect_nonexistent_testdata `git diff --exit-code $$(git merge-base master HEAD) | grep '+  !include: test_data/' | grep 'yaml$$' | awk '{print $$3}'`
```

Comment:
I think we are missing here to check !include in `schedule/`

Reply:
The grep '+  !include: test_data/' will grep lines in git diff, the file which contains the lines can be in schedule/, there is no location limitation.

Reply:
Ahm! my fault, I didn't read it properly because I was expecting something else. The thing is that I was checking the parser and we don't force test_data/ as a folder for test_data, so for example for unit test would not be detected, but as that folder is special I think it doesn't matter and I was checking for other teams using other folder and seems they are not, so it should be fine, it is the right expectation.
Where I spot that it didn't match anything is when having:
```
- !include: 
    - path/to/first_test_data.yaml
    - path/to/second_test_data.yaml
```
When reading it, I think it is not required the `-` in front of !include (it is parsed in line 125 in lib/scheduler.pm` because the value is a arrayref and the key doesn't matter if it is an array. So we could double-check that works with:
```
!include: 
    - test_data/to/first_test_data.yaml
    - test_data/second_test_data.yaml
```
and see if it worths to add the CI validation here too or not.

Reply:
> Ahm! my fault, I didn't read it properly because I was expecting something else. The thing is that I was checking the parser and we don't force test_data/ as a folder for test_data, so for example for unit test would not be detected, but as that folder is special I think it doesn't matter and I was checking for other teams using other folder and seems they are not, so it should be fine, it is the right expectation.

but if I don't distinguish yaml files according to location, what other characteristic determines it is a test data file and not a schedule or other yaml file?

> !include: 
>     - test_data/to/first_test_data.yaml
>     - test_data/second_test_data.yaml
> ```

I think that can be fixed by a regex, I will try to include this scenario.

Reply:
only the `!include` keyword, after that there is a path or a list of path.

Reply:
I believe we should also check if `TEST_DATA` setting is given in the variables, not only include.

Reply:
those settings are from running job, I think we are only covering static check with this. If we want to catch that then I guess we should try from GitLab job group yaml or even from the test suite in openQA web ui, but if we move all what wee need to job group, we can apply the CI there, make sense? https://progress.opensuse.org/issues/64749

Reply:
> only the `!include` keyword, after that there is a path or a list of path.

but in case of delete/rename we don't have that info in the pull request.

>I believe we should also check if TEST_DATA setting is given in the variables, not only include.

How can I do that? Isn't there the possibility that the variable will not be set, but the schedule will include test_data anyway?

>    !include:
>    - test_data/to/first_test_data.yaml
>    - test_data/second_test_data.yaml

I can use something like "grep -zP '\include: \n' but I could not find a single example in the repo. Is this really a possible scenario?

Reply:
it is possible, because we allow that in the parser as I mentioned before, but there is no use case for now of that, so I think you investigated enough, thanks for that, maybe if we need it in the future we can tweak this script. I would resolve this conversation if you don't mind.